var/spack/repos/builtin/packages/grid/package.py

```diff
@@ -0,0 +1,149 @@
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+
+    depends_on('gmp')
+    depends_on('mpfr')
+
+    variant('comms', default='mpi',
+            values=('none', 'mpi', 'mpi3', conditional('shmem', when='^cray-mpich')),
+            description='Choose communication interface')
+    depends_on('mpi', when='comms=mpi')
+    depends_on('cray-mpich', when='comms=shmem')
+    depends_on('mpi@3:', when='comms=mpi3')
+
+    variant('fftw', default=True, description='Activate FFTW support')
+    depends_on('fftw-api@3', when='+fftw')
+
+    variant('lapack', default=False, description='Activate LAPACK support')
+    depends_on('lapack', when='+lapack')
+
+    variant('hdf5', default=False, description='Activate HDF5 support')
+    depends_on('hdf5', when='+hdf5')
+
+    variant('lime', default=False, description='Activate LIME support')
+    depends_on('c-lime', when='+lime')
+
+    variant('doc', default=False,
+            description='Build the documentation with doxygen')
+    depends_on("doxygen", type="build", when="+doc")
+
+    variant('gen-simd-width', default='64',
+            description='Size (in bytes) of the generic SIMD vector type')
+    variant('rng', default='sitmo', values=('sitmo', 'ranlux48', 'mt19937'),
+            multi=False, description='RNG setting')
+    variant('timers', default=True,
+            description='System dependent high-resolution timers')
+    variant('chroma', default=False,
+            description='Chroma regression tests')
+
+    def autoreconf(self, spec, prefix):
+        Executable('./bootstrap.sh')()
+
+    def configure_args(self):
+        spec = self.spec
+        args = ['--with-gmp', '--with-mpfr']
+
+        if spec.satisfies('^intel-mkl'):
+            if '+fftw' in spec or '+lapack' in spec:
+                args.append('--enable-mkl')
+        else:
+            if '+fftw' in spec:
+                args.append('--with-fftw={0}'.format(self.spec['fftw'].prefix))
+            if '+lapack' in spec:
+                args.append('--enable-lapack={0}'.format(self.spec['lapack'].prefix))
+                # lapack is searched only as `-llapack`, so anything else
+                # wouldn't be found, causing an error.
+                args.append('LIBS={0}'.format(self.spec['lapack'].libs.ld_flags))
+
+        if 'comms=none' not in spec:
+            # The build system can easily get very confused about MPI support
+            # and what linker to use.  In many case it'd end up building the
+            # code with support for MPI but without using `mpicxx` or linking to
+            # `-lmpi`, wreaking havoc.  Forcing `CXX` to be mpicxx should help.
+            args.extend([
+                "CC={0}".format(spec['mpi'].mpicc),
+                "CXX={0}".format(spec['mpi'].mpicxx),
+            ])
+
+        if '+timers' in spec:
+            args.append('--enable-timers')
+        else:
+            args.append('--disable-timers')
```

Comment:
Here you can:
```suggestion
        args += self.enable_or_disable('timers')
```
Similarly for the variants below.

Reply:
Cool!